File: android/app/src/main/java/com/parkingkok/app/domain/detection/DetectionState.kt
package com.parkingkok.app.domain.detection

/**
 * Shared detection state machine states.
 * Semantics are fixed by docs/05_CROSS_PLATFORM_DOMAIN_CONTRACT.md §3 and must stay
 * identical to the iOS enum. Transition logic itself is out of M0B-1 scope.
 */
enum class DetectionState {
    IDLE,
    DRIVING_CANDIDATE,
    DRIVING,
    PARKING_CANDIDATE,
    PARKED,
    DEPARTURE_CANDIDATE,
}

File: ios/ParkingKok/Detection/Domain/DetectionState.swift
import Foundation

/// The shared detection state vocabulary
/// (docs/05_CROSS_PLATFORM_DOMAIN_CONTRACT.md §3).
///
/// Raw values are the contract strings so a checkpoint written by this build stays
/// readable — and comparable against the Android implementation — without a mapping
/// table. The transition rules that move between these states are M0A-2; this
/// milestone only needs to persist and restore the value.
///
/// Note: `docs/05_PARKING_DETECTION_ENGINE.md` §3 lists `PARKING_TRANSITION` and
/// `CANDIDATE_PENDING` where the cross-platform contract lists `PARKING_CANDIDATE`.
/// CLAUDE.md ranks the contract above the engine spec, so the contract wins here.
enum DetectionState: String, Sendable, Codable, CaseIterable {
    case idle = "IDLE"
    case drivingCandidate = "DRIVING_CANDIDATE"
    case driving = "DRIVING"
    case parkingCandidate = "PARKING_CANDIDATE"
    case parked = "PARKED"
    case departureCandidate = "DEPARTURE_CANDIDATE"
}
